ts/torch_handler/base_handler.py

```diff
@@ -44,14 +44,12 @@ def initialize(self, context):
 
         """
         properties = context.system_properties
-        self.map_location = "cuda" if torch.cuda.is_available() else "cpu"
-        self.device = torch.device(
-            self.map_location + ":" + str(properties.get("gpu_id"))
-            if torch.cuda.is_available()
-            else self.map_location
-        )
-        self.manifest = context.manifest
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        if torch.cuda.is_available():
+            torch.cuda.set_device(properties.get("gpu_id"))
```

Comment:
just an FYI:
If I'm running torchserve gpu version(in which case torch.cuda.is_available returns True) and choose to use `number_of_gpu=0` (temporarily for whatever reason)
`gpu_id` is set to None and it errors out because of "cuda:None"
before assigning device we should also check `gpu_id is not None`
I was gonna open a separate issue last week but maybe we can just solve it here?~
As @ngimel points out; "cuda" without id is suspect

Reply:
Thanks, @cceyda we will check for this issue as well and good point, adding a check for gpu_id is needed as well.